Haskell updates

domenkozar opened the pull request.

peti committed:
haskell-liquidhaskell: drop obsolete overrides

TravisWhitaker committed:
GHCJS: Build on Hydra

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.0 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/43ab24038ab24393941e9cc4f06e628dd9183411.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.0 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/05d7197f3555ac17eb38c7a64344c906d8360029.

mdorman committed:
cabal2nix: fix build

Without this change, the build fails with:

```
Warning:
    This package indirectly depends on multiple versions of the same package. This is very likely to cause a compile failure.
      package hackage-db (hackage-db-2.0.1-AliTRXEJBGFAEMpONOEnbV) requires Cabal-2.4.0.1
      package hpack (hpack-0.32.0-1iJuukTWmC4EGkYhAiFeII) requires Cabal-3.0.0.0-CD1INMNwoxXIKVgcu1nec5
      package distribution-nixpkgs (distribution-nixpkgs-1.3.1-82Z2DCdLAUs8Q2gjtlPLz5) requires Cabal-3.0.0.0-CD1INMNwoxXIKVgcu1nec5
      package cabal2nix (cabal2nix-2.15.0) requires Cabal-3.0.0.0-CD1INMNwoxXIKVgcu1nec5
      package cabal2nix (cabal2nix-2.15.0) requires Cabal-3.0.0.0-CD1INMNwoxXIKVgcu1nec5
      package cabal2nix (cabal2nix-2.15.0) requires Cabal-3.0.0.0-CD1INMNwoxXIKVgcu1nec5
```

peti committed:
Merge pull request #69640 from mdorman/cabal2nix-fix

cabal2nix: fix build

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.0 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/47b2f92d3c7f9fc506ca0534b68b765b25c3aa00.

nmattia committed:
niv: mark as not broken

`niv` used to depend on the broken `github`. The dependency was removed
and `niv` builds fine.

peti committed:
Merge pull request #69441 from TravisWhitaker/hydra-ghcjs

GHCJS: Build on Hydra

peti committed:
Merge pull request #69870 from nmattia/nm-unbreak-niv

niv: mark as not broken

domenkozar commented:
@GrahamcOfBorg build cachix niv cabal2nix

domenkozar committed:
fix dhall_1_26_0 reference

domenkozar commented:
@GrahamcOfBorg eval

peti reviewed (changes requested):
:-1: 

This branch is in no shape for a merge. Some of the changes to `configuration-hackage2nix.yaml` didn't even have a chance to take effect yet because the generator wasn't run since they were added.

domenkozar commented:
Sure, I'd like to understand what's missing and help out. I should have opened a draft PR, but too late :)

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.0 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/04348dc32c9d89ce4eda31f13805def85a89857f.

domenkozar committed:
haskell.packages.ghc881.regex-posix: fix build

domenkozar committed:
cachix&niv: use separate bin output

domenkozar commented:
@GrahamcOfBorg build cachix niv cabal2nix

domenkozar committed:
haskellPackages.amazonka*: fix build

domenkozar commented:
@peti https://hydra.nixos.org/eval/1545587 looks good

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.0 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/fd0a3e82ed9baf9ccd250ad6c1c8bccd950ac9f4.

peti commented on pkgs/development/haskell-modules/configuration-nix.nix:
```diff
@@ -94,11 +94,9 @@ self: super: builtins.intersectAttrs super {
-      substituteInPlace cachix.cabal --replace "c++14" "c++17"
-    '';
-  });
+  cachix = addBuildTools (enableSeparateBinOutput super.cachix) [pkgs.boost];
```
What part of `boost` does `cachix` need? Could you please document that in a comment?

domenkozar replied:
See 821676200d6f65c9eb4584e393af1204eeb7e363

peti committed:
haskell-hlint: update override to use the latest ghc-lib-parser version

peti committed:
haskell-ghc-lib-parser: we mo longer need older versions around

peti committed:
hackage2nix: disable failing builds that break evaluation on Hydra

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.0 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/65cacc5a5a1ef0ecdbb1d582b3cccc82ba00727c.

domenkozar committed:
cachix: comment why boost is needed

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.0-1-g5325ccd from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/48972320ba180459d6ad28ab4f21389b1b3d7fc6.

peti committed:
cachix: it's no longer necessary to add boost manually

domenkozar commented:
@peti what's left?

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.0-1-g5325ccd from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/36c258915faffff2b555a49d04b4905c8034c2bd.

peti commented:
> @peti what's left?

Haddock is currently broken.

peti reviewed (changes requested):
We need a working haddock, I suppose: https://hydra.nixos.org/eval/1546016

domenkozar commented:
OK. I'm going to make another release of cachix and we can merge this tomorrow.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.0-1-g5325ccd from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/c3176c037fe0413020074aeda2569fdace1378cb.

domenkozar commented:
Released, if you update the set we can fix haddock and merge.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.0-1-g5325ccd from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/b51789ef3eed6bcb3a786ae581d424a62526e333.

domenkozar commented:
```
diff --git a/pkgs/development/haskell-modules/configuration-hackage2nix.yaml b/pkgs/development/haskell-modules/configuration-hackage2nix.yaml
index 70dddd52252..4604a941ae8 100644
--- a/pkgs/development/haskell-modules/configuration-hackage2nix.yaml
+++ b/pkgs/development/haskell-modules/configuration-hackage2nix.yaml
@@ -2488,11 +2488,7 @@ extra-packages:
   - gloss < 1.9.3                       # new versions don't compile with GHC 7.8.x
   - haddock < 2.17                      # required on GHC 7.10.x
   - haddock == 2.17.*                   # required on GHC 8.0.x
-  - haddock-api == 2.15.*               # required on GHC 7.8.x
-  - haddock-api == 2.16.*               # required on GHC 7.10.x
-  - haddock-api == 2.17.*               # required on GHC 8.0.x
-  - haddock-library == 1.2.*            # required for haddock-api-2.16.x
-  - haddock-library == 1.4.3            # required for haddock-api-2.17.x
+  - haddock-api == 2.22.*               # 2.23.* breaks on GHC 8.6
   - happy <1.19.6                       # newer versions break Agda
   - haskell-gi-overloading == 0.0       # gi-* packages use this dependency to disable overloading support
   - haskell-src-exts == 1.19.*          # required by hindent and structured-haskell-mode
```

This should fix it I guess.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.0-1-g5325ccd from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/845035f8d911809290588c5ca4b6e3b5eb458f90.

domenkozar commented:
@peti does the above patch look good?

peti commented:
> @peti does the above patch look good?

Yes, it looks good to me. I'm not sure whether other haddock packages will build properly with a version of `haddock-api` that's older than their own ... but they might. I haven't checked.

domenkozar committed:
haddock-*: fix by pinning

domenkozar commented:
@peti cool, if you can regenerate the packages I'll verify they build and we can merge.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.0-1-g5325ccd from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/54d61d70bc071d03e652e6d662e4d944d20723c5.

peti commented:
> Ff you can regenerate the packages I'll verify they build and we can merge.

Done. In case you are not aware of the video, https://www.youtube.com/watch?v=qX0mgtSm360 might be of interest.

domenkozar committed:
haddock: fix build

domenkozar commented:
```
$ nix-build -A haskellPackages.haddock
/nix/store/whfsdjijy0arma080lw8s087gl11pbsy-haddock-2.22.0
```

It seems to work again. Going to merge after CI passes the evaluation checks.

peti commented:
I would not have merged the branch in this state. I would have liked to do more clean-up first. :disappointed:

domenkozar commented:
I really wanted nixpkgs master fixed, seems like the cleanup can be done on top of this work? :)

peti commented:
Well, if you don't want to take my interests seriously, then I guess I'll just file that under "experience".

domenkozar commented:
@peti I've asked you (multiple times) in this PR:

- that I'd really like to get this merged
- what do you think is missing

You approved the PR, then said we have to fix haddock. I've shown you the diff I applied (which I simplified).

So I'm quite certain I did my best to give you a chance to communicate what you'd like to do.

domenkozar commented:
That said, thanks for the video, I'll watch it.
